hls-plugin-api/src/Ide/PluginUtils.hs

```diff
@@ -243,6 +246,14 @@ allLspCmdIds pid commands = concatMap go commands
 
 -- ---------------------------------------------------------------------
 
+getNormalizedFilePath :: Monad m => PluginId -> TextDocumentIdentifier -> ExceptT String m NormalizedFilePath
```

Comment:
i guess something like this is being used everywhere and a nice refactoring would be use it (forother pr)

Reply:
Read my mind!!!